src/org/zaproxy/zap/extension/csphelper/CspHelperPassiveScanner.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.csphelper;
+
+import java.util.List;
+import java.util.Random;
+
+import net.htmlparser.jericho.Element;
+import net.htmlparser.jericho.Source;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+import org.zaproxy.zap.model.Vulnerabilities;
+import org.zaproxy.zap.model.Vulnerability;
+
+/**
+ * An example passive scan rule, for more details see 
+ * http://zaproxy.blogspot.co.uk/2014/04/hacking-zap-3-passive-scan-rules.html
+ * @author psiinon
+ */
+public class CspHelperPassiveScanner extends PluginPassiveScanner {
```

Comment:
Implement `PassiveScanner` instead (e.g. [ParamScanner](https://github.com/zaproxy/zaproxy/blob/a84c98fbb6b97ff5400539295df0091202fb299b/src/org/zaproxy/zap/extension/params/ParamScanner.java)). Note that in this case you have to manually add/remove it with the extension (e.g. [ExtensionParam](https://github.com/zaproxy/zaproxy/blob/a84c98fbb6b97ff5400539295df0091202fb299b/src/org/zaproxy/zap/extension/params/ExtensionParams.java#L131-L150)).